src/main/java/net/snowflake/ingest/streaming/internal/ArrowFlusher.java

```diff
@@ -94,6 +95,7 @@ public Flusher.SerializationResult serialize(
         // Write channel data using the stream writer
         arrowWriter.writeBatch();
         rowCount += data.getRowCount();
+        chunkUncompressedSize += data.getBufferSize();
```

Comment:
chunkData.size and the accumulation of buffer size could be different, right? I think the buffer size is the estimation  and we should use the actual uncompressed data size?

Reply:
I think it is fine to use the estimation. We use here in other places as well, eg this [log line](https://github.com/snowflakedb/snowflake-ingest-java/pull/489/files#diff-0b5d892d276fc68717bae6286a9fc43ae7ff3d033b0e260c66e781179c7143ddL60) and this [other](https://github.com/snowflakedb/snowflake-ingest-java/pull/489/files#diff-0b5d892d276fc68717bae6286a9fc43ae7ff3d033b0e260c66e781179c7143ddL102).

Reply:
We can actually get back to the original behaviour for Arrow if `SerializationResult` had getUncompressedSize() method overriden for Arrow and Parquet accordingly or only for parquet.

Reply:
Arrow will be deprecated soon anyway.